force-app/main/default/classes/ContactController.cls

```diff
@@ -1,170 +1,132 @@
+    private static List<Id> getAllContactsId(List<Contact> delContacts, Map<Id, List<Contact>> accountWithContacts){
+        List<Id> result = new List<Id>();
+        List<Contact> tempContacts = new List<Contact>();
+        for(List<Contact> temp : accountWithContacts.values()){
+            tempContacts.addAll(temp);
+        }
+        tempContacts.addAll(delContacts);
+        for(Contact c : tempContacts){
+            result.add(c.Id);
         }
         return result;
-    }   
+    }
 
-    private static Contact findContactWithMinCases (Id contactId){        
-        List<Contact> source = findContactColleagues(contactId);        
-        if (source.size() == 0) {
-            return null;
-        }else {   
-            Contact result;  
-            Map<Id, Contact> contactsWithCountCases = new Map<Id, Contact>(source);
-            List<Contact> resList = [SELECT Id, (SELECT Id FROM Cases) FROM Contact WHERE Id IN :contactsWithCountCases.keySet()];
-            Map<Contact, Integer> resMap = new Map<Contact, Integer>();
-            for(Contact c : resList){
-                resMap.put(c, c.Cases.size());
-            }           
-            List<Contact> contactList = new List<Contact>(resMap.keySet());
-            List<Integer> caseQuantity = resMap.values();
-            Integer count = 2147483646;
-            Integer tmp;
-            for(Integer i = 0; i < caseQuantity.size(); i++){
-                if (caseQuantity.get(i) < count) {
-                    count = caseQuantity.get(i);
-                    tmp = i;
-                }
+    private static Map<Contact, List<Case>> getAllCases(List<Id> allContactsId){
+        List<Contact> allContacts = [SELECT Id, AccountId, (SELECT Id, Status, ContactId FROM Cases) FROM Contact WHERE Id IN :allContactsId];
+        Map<Contact, List<Case>> result = new Map<Contact, List<Case>>();
+        for(Contact c : allContacts){
+            result.put(new Contact(Id = c.Id, AccountId = c.AccountId), c.Cases);
+            if(c.Cases.size() > maxCases){
+                maxCases = c.Cases.size();
             }
-            result = contactList.get(tmp);
-           return result;
-        }        
+        }            
+        return result;
     }
-}
\ No newline at end of file
+
+    private static Map<Contact, List<Case>> getSpecificContactsWithCases(Boolean check, Map<Contact, List<Case>> source, List<Contact> delContacts){
+        Map<Contact, List<Case>> result = new Map<Contact, List<Case>>();
+        Map<Contact, List<Case>> del = new Map<Contact, List<Case>>();
+        Map<Contact, List<Case>> stay = new Map<Contact, List<Case>>();
+        for(Contact c : source.keySet()){
+            if (delContacts.contains(c)) {
+                del.put(c, source.get(c));
+            }else {
+                stay.put(c, source.get(c));
+            }            
+        }
+        if (check) {
+            result = del;
+        }else {
+            result = stay;
+        }
```

Comment:
it would be more elegant to use ternary operator here, like:
`result = check ? del : stay;`

Reply:
Fixed